### PR TITLE
Optimize aggregate

### PR DESCRIPTION
**Pull request by jnation3406**
I tried a bunch of things, including implementing raw sql for a loose indexscan (https://wiki.postgresql.org/wiki/Loose_indexscan), and putting and index on the individual fields and this group index. The group index by far helped the most, as well as this other small optimization I found. The loose indexscan for some reason helped a bit on a few of the fields, but was way slower on a few others (proposal_id, and instrument_id).

This will require a migration that I will add later once I reset the db. The index together migration only took a minute to apply and added ~2GB to the total db size, which is about 1%. This dropped the time of doing an ALL aggregation from ~35s to ~4s, and each individual aggregate_field aggregation from ~6-7s to a little under 1s.

### CODE REVIEW

File: archive/frames/views.py [hunk 1]
```diff
@@ -154,13 +154,14 @@ def zip(self, request):
         return Response(request_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
     @staticmethod
-    def _get_aggregate_values(query_set, field, aggregate_field):
+    def _get_aggregate_values(query_set, query_filters, field, aggregate_field):
+        if field in query_filters:
+            return [query_filters[field]]
```
Comment (jnation3406) [inline archive/frames/views.py:159]: This just ensures we dont waste time doing an aggregate query when we are filtering by that specific term too.

**Review by mgdaily — APPROVED** [review]
Nice!!

**Review by jashan-lco — APPROVED** [review]
LGTM.

I'd still be curious to see how doing the aggregation in one query using https://docs.djangoproject.com/en/4.0/ref/contrib/postgres/aggregates/#arrayagg would affect the performance.

**Comment by jnation3406** [comment]
> LGTM.
> 
> I'd still be curious to see how doing the aggregation in one query using https://docs.djangoproject.com/en/4.0/ref/contrib/postgres/aggregates/#arrayagg would affect the performance.

I'm waiting on Jesus to kill the db so I can re-clone and try it all again (since I did multiple other migrations in it during testing) before deploying, but once I get the new database going I can try testing this as well.

**Pull request merged**